libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1911,8 +1911,26 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 			break;
 		}
 		case Type::Category::Struct:
-			solUnimplementedAssert(false, "Struct conversion not implemented.");
+		{
+			solAssert(toCategory == Type::Category::Struct, "");
+			auto const& fromStructType = dynamic_cast<StructType const &>(_from);
+			auto const& toStructType = dynamic_cast<StructType const &>(_to);
+			solAssert(fromStructType.structDefinition() == toStructType.structDefinition(), "");
+
+			solUnimplementedAssert(!fromStructType.isDynamicallyEncoded(), "");
+			solUnimplementedAssert(toStructType.location() == DataLocation::Memory, "");
+			solUnimplementedAssert(fromStructType.location() == DataLocation::CallData, "");
+
+			Whiskers t(R"(
+				converted := <abiDecode>(value, calldatasize())
+			)");
+			t("abiDecode", ABIFunctions(m_evmVersion, m_revertStrings, m_functionCollector).tupleDecoder(
+				{&toStructType}
+			));
+
+			body = t.render();
 			break;
```

Comment:
Could be contracted to:
```suggestion
			body = Whiskers t(R"(
				converted := <abiDecode>(value, calldatasize())
			)")("abiDecode", ABIFunctions(m_evmVersion, m_revertStrings, m_functionCollector).tupleDecoder(
				{&toStructType}
			)).render();
			break;
```
But fine either way!

Reply:
Ah, as I have another PR (will open it soon) that is rebased on top of this, I did this change actually there. So I think will do it here, as it is naturally part of this one